packages/playwright-core/src/server/injected/recorder/recorder.ts

```diff
@@ -1039,7 +1039,7 @@ export class Recorder {
 
     this.highlight.install();
     // some frameworks erase the DOM on hydration, this ensures it's reattached
-    const recreationInterval = setInterval(() => {
+    const recreationInterval = this.injectedScript.builtinSetTimeout(() => {
```

Comment:
This should be setInterval. Or call setTimeout recursively.

Reply:
good catch!